barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.cpp

```diff
@@ -124,6 +127,7 @@ std::shared_ptr<typename ProverInstances::Instance> ProtoGalaxyProver_<ProverIns
         combined_relation_parameters.lookup_grand_product_delta.evaluate(challenge),
     };
     next_accumulator->relation_parameters = folded_relation_parameters;
+    next_accumulator->prover_polynomials = ProverPolynomials(next_accumulator->proving_key);
```

Comment:
derive the prover_polynomials from proving_key (very cheap step of .share() and .shifted() calls)

Reply:
can you add a comment above explaining  why this is happening?